### ISSUE TITLE
Some query rules don't seem supported

### ISSUE THREAD

**Issue by julienbourdeau** [algolia/algoliasearch-client-java#514]
- Algolia Client Version: latest
- Language Version: any

### Description

I'm trying to add the following query rule via the Java Client.

```json
{
  "condition": {
    "pattern": "product",
    "anchoring": "contains"
  },
  "consequence": {
    "params": {
      "query": {
        "remove": [
          "product"
        ]
      }
    }
  },
  "objectID": "1532332845427"
}
```

I can't because i'm unable to instantiate the `ConsequenceQuery` object. I don't understand why it was made abstract, but there could be a reason I miss.

```java
public abstract class ConsequenceQuery implements Serializable {}
```

https://admin.algolia.com/admin/users/57646/applications/EKVLW9VMCB/graphs/perf

The same way, if you create it via the dashboard and try to export the rules, it breaks.

## Step to reproduce

You can reproduce the issue by adding a query rule in the dashboard:

```json
{
  "condition": {
    "pattern": "product",
    "anchoring": "contains"
  },
  "consequence": {
    "params": {
      "query": {
        "remove": [
          "product"
        ]
      }
    }
  },
  "objectID": "1532332845427"
}
```

Run this code (or similar, the important part is the export)

```java
public class RulesAction {
    private APIClient client;
    private Index<?> index;
    
    private final String ID = "testingMEF4K8CU83";
    private final String KEY = "c617e93a43c632c2c17d5e187bda8a39";
    private final String INDEX = "test_query_rules";

    public RulesAction() {
        this.client = new ApacheAPIClientBuilder(ID, KEY).build();
        this.index = this.client.initIndex(INDEX);
    }
    
    public static void main(String[] args) {
        RulesAction exporter = new RulesAction();
        exporter.exportRules();
    }
    
    private void exportRules() {
        RulesIterable rules = new RulesIterable(this.index, 1);
        for(Rule rule : rules )
            System.out.println(rule.getObjectID());
                
    }
}
```

You'll get

![a263](https://user-images.githubusercontent.com/1525636/47073160-ed1c9200-d1f7-11e8-9693-97e12c9b40a6.png)

**Comment by aseure** [comment]
As I told @julienbourdeau, there's certainly an deserialisation issue there but not with the rule that was provided. I've tried to set it, get it, search it and iterate over the rules, but got not exception at all. So the rule with which they seem to have an error is probably not this one.

I'm currently going back to the original discussions to find the original rule and code sample and will post it here if I can find it.

**Comment by aseure** [comment]
<img width="1349" alt="screenshot 2018-10-17 at 16 54 01" src="https://user-images.githubusercontent.com/3115191/47095664-f3c5fc00-d22d-11e8-91cd-7adcb418b226.png">

**Comment by aseure** [comment]
Actually already fixed on https://github.com/algolia/algoliasearch-client-java-2/commit/d6b0e83e88c2690146f95215af1bcfaad2dca6d7 but not yet released. Will be available as of 2.20.0 that we will release today.
